continuationsPlugin/src/test/scala/continuations/ContinuationsPluginSuite.scala

```diff
@@ -706,4 +709,67 @@ class ContinuationsPluginSuite extends FunSuite, CompilerFixtures {
           assertNoDiff(compileSourceIdentifier.replaceAllIn(tree.show, ""), expected)
       }
   }
+
+  compilerContextWithContinuationsPlugin.test(
+    "it should convert simple nested suspended def with a single parameter keeping the rest of the rows untouched") {
+    case given Context =>
+      val source =
+        """|
+           |package continuations
+           |
+           |def foo(x: Int)(using Suspend): Int = {
+           |  val y = 5
+           |  println("HI")       
+           |  if (x < y) then {
+           |    x
+           |  } else {
+           |    summon[Suspend].suspendContinuation[Int] { continuation => continuation.resume(Right(1)) }
+           |  }
+           |}
+           |""".stripMargin
+
+      // format: off
+      val expected =
+        """|
+           |package continuations {
+           |  final lazy module val compileFromString$package: 
+           |    continuations.compileFromString$package
+           |   = new continuations.compileFromString$package()
+           |  @SourceFile("compileFromString.scala") final module class 
+           |    compileFromString$package
+           |  () extends Object() { this: continuations.compileFromString$package.type =>
+           |    private def writeReplace(): AnyRef = 
+           |      new scala.runtime.ModuleSerializationProxy(classOf[continuations.compileFromString$package.type])
+           |    def foo(x: Int, completion: continuations.Continuation[Int]): Any | Null | continuations.Continuation.State.Suspended.type = 
+           |      {
+           |        val y: Int = 5
+           |        println("HI")
+           |        if x.<(y) then 
+           |          {
+           |            x
+           |          }
+           |         else 
+           |          {
+           |            {
```

Comment:
The continuation is inserted inside a `Block` tree, I don't know if this causes any problem or how to add the new continuation `ValDef`s  without being inside a block

Reply:
I guess the "issue" is from [here](https://github.com/47deg/TBD/pull/60/files#diff-fade65ebf959155389570d62d274801c2f7607f965a273661e5f776636d6393fR148). if we use `tpd.Thicket` instead of `Block` does it change anything?

also more general out-of-curiosity question, does this now work with a `suspendContinuation` call nested in deeper `ifs`?

Reply:
Exactly, I mean if it was a problem to have the definitions inside a different `Block`, instead of being at the same level as the rest of the statements of that level (if there are any).

Good suggestion about using `Thicket`! I'll check.

Yes, now it should work for any level of nesting 😄

Reply:
>(if there are any)

yeah you are right can't think of something rn. so probably can ignore it 

>Yes, now it should work for any level of nesting

🎉

Reply:
Blocks don't matter in my opinion. Let's not get hung up on the forest for the trees. Other transforms use them liberally - Closure for example.